### ISSUE TITLE
Inconsistent? Handling of whitespace in keys, at least in redis-cli

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#197]
```
What steps will reproduce the problem?

$ ./redis-cli set foo bar
OK
$ ./redis-cli get foo
bar
$ ./redis-cli set "quoted-foo" "quoted-bar"
OK
$ ./redis-cli get "quoted-foo"
quoted-bar
$ ./redis-cli get quoted-foo
quoted-bar
$ ./redis-cli set "foo witt whitespace" "bar with whitespace"
(error) ERR wrong number of arguments for 'set' command
$ ./redis-cli get "foo witt whitespace"
(error) ERR wrong number of arguments for 'get' command
$ ./redis-cli set "foo-with-trailing-whitespace     " "bar with whitespace"
OK
$ ./redis-cli get "foo-with-trailing-whitespace     "
bar with whitespace
$ ./redis-cli get foo-with-trailing-whitespace
bar with whitespace
$ ./redis-cli set "    foo-with-leading-whitespace" "bar with whitespace"
OK
$ ./redis-cli get "    foo-with-leading-whitespace"
bar with whitespace
$ ./redis-cli get "foo-with-leading-whitespace"
bar with whitespace

On branch master:

$ git log -1
commit b3f83f127273da21506f697e256ae010587b10f1
Author: Jeremy Zawodny <Jeremy@Zawodny.com>
Date:   Fri Mar 19 22:24:17 2010 -0700

    support for include directive in config parser

```

Original issue reported on code.google.com by `anibalro...@gmail.com` on 20 Mar 2010 at 12:35

**Comment by GoogleCodeExporter** [comment]
```
Ups!Sorry it slipped as accepted, not sure what the behaviour should be. I 
remember a
discussion about whitespaces in keys, but I am not sure what happened in the 
end.
Anyway the behaviour is inconsistent IMHO ;)
```

Original comment by `anibalro...@gmail.com` on 20 Mar 2010 at 12:37

**Comment by GoogleCodeExporter** [comment]
```
I use the following to get around invalid-key characters. (hex of ascii)

" " -> %20
"(" -> %28
")" -> %29
"[" -> %5b
"]" -> %5d

helper method does a replace on incoming and outgoing keys based on this 
replacement
pattern.

```

Original comment by `kata...@gmail.com` on 15 Apr 2010 at 2:47

**Comment by GoogleCodeExporter** [comment]
```
This is now completely fixed, since Redis 2.0:


redis> set "any kind of key\n" foo
OK
redis> get "any kind of key\n"
"foo"

Closing :)
Salvatore
```

Original comment by `anti...@gmail.com` on 27 Aug 2010 at 11:02
- Changed state: **Verified**
